pkg/typesystem/typesystem.go

```diff
@@ -345,6 +345,26 @@ func (t *TypeSystem) UsersetCanFastPath(relationReferences []*openfgav1.Relation
 	return true
 }
 
+func RelationEquals(a *openfgav1.RelationReference, b *openfgav1.RelationReference) bool {
+	if a.GetType() == b.GetType() {
+		// Type with no relation or wildcard (e.g. 'user').
+		if a.GetRelationOrWildcard() == nil && b.GetRelationOrWildcard() == nil {
+			return true
+		}
+
+		// Typed wildcard (e.g. 'user:*').
+		if a.GetWildcard() != nil && b.GetWildcard() != nil {
+			return true
+		}
+
+		if a.GetRelation() != "" && b.GetRelation() != "" &&
+			a.GetRelation() == b.GetRelation() {
+			return true
+		}
+	}
+	return false
```

Comment:
**(suggestion):** We could further reduce complexity by inverting these ifs. 
```suggestion
	if a.GetType() != b.GetType() {
		return false
	}

	// Type with no relation or wildcard (e.g. 'user').
	if a.GetRelationOrWildcard() == nil && b.GetRelationOrWildcard() == nil {
		return true
	}

	// Typed wildcard (e.g. 'user:*').
	if a.GetWildcard() != nil && b.GetWildcard() != nil {
		return true
	}

	return a.GetRelation() != "" && b.GetRelation() != "" && a.GetRelation() == b.GetRelation()
```

Reply:
This is a good point, and would be an easy win. However, for the purposes of this PR example, I was attempting to minimize logic changes and focus primarily on testing smaller units of work.